### PR TITLE
WasReqUrl tests for Social Clients

### PR DESCRIPTION
**Pull request by c00crane**
Create WasReqUrl tests for social clients.
I'll use the OIDC client tests - will share those between OIDC and Social clients.
I'll move the test cases as well as the OP config from the com.ibm.ws.security.oidc.client_fat.2 project to com.ibm.ws.security.fat.common.SSO.clientTests.
I'm moving the RP config to the com.ibm.ws.security.oidc.client_fat.3 project.  I'll remove OidcClientWasReqURLTests.class from the com.ibm.ws.security.oidc.client_fat.2 FATSuite and add it to the com.ibm.ws.security.oidc.client_fat.3 FATSuite.
I'll create a LibertyOP_Social_ClientWasReqURLTests class in com.ibm.ws.security.social_fat.LibertyOP.3 and add it to the FATSuite of that project.  I'll also add a social client config (equivalent to the RP client config used by the OIDC client).

#25734

### CODE REVIEW

**Comment by c00crane** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.social_fat.LibertyOP.3
(Note:  waiting on a couple runtime fixes, so com.ibm.ws.security.social_fat.LibertyOP.3 will have failures at this time - running tests now to make sure that there are no unexpected failues)

**Comment by c00crane** [comment]
#libby
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.2,com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.social_fat.LibertyOP.3
Running tests with the WasReqUrl tests disabled for social clients - will enable those once we have the runtime fixes.

**Comment by c00crane** [comment]
This delivery needs to wait on pulls #27994 & #27993 as they resolve problems found by these tests.

**Comment by c00crane** [comment]
#libby
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.social_fat.LibertyOP.3

**Comment by c00crane** [comment]
https://github.com/OpenLiberty/open-liberty/pull/27994 & https://github.com/OpenLiberty/open-liberty/pull/27993 have been delivered.

File: dev/com.ibm.ws.security.fat.common.SSO.clientTests/publish/servers/com.ibm.ws.security.SSO.op.wasReqUrl/bootstrap.properties [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+com.ibm.ws.security.oauth*=all=enabled:\
+com.ibm.ws.security.openidconnect*=all=enabled:\
+com.ibm.ws.security.jwt*=all=enabled:\
+com.ibm.ws.security.common*=all=enabled:\
+com.ibm.ws.webcontainer.security.*=all=enabled:\
+com.ibm.oauth.*=all=enabled:\
+com.ibm.wsspi.security.oauth20.*=all=enabled:\
+org.apache.http.client.*=all:\
+OpenIdConnect=all:\
+OPENIDCONNECT=all
```
Comment (ayoho) [inline dev/com.ibm.ws.security.fat.common.SSO.clientTests/publish/servers/com.ibm.ws.security.SSO.op.wasReqUrl/bootstrap.properties:28]: Duplicate of lines above. Also could use to add `io.openliberty.security.*=all`.

File: dev/com.ibm.ws.security.fat.common.SSO.clientTests/publish/servers/com.ibm.ws.security.SSO.op.wasReqUrl/bootstrap.properties [hunk 2]
```diff
@@ -0,0 +1,31 @@
+###############################################################################
+# Copyright (c) 2021 IBM Corporation and others.
```
Comment (ayoho) [inline dev/com.ibm.ws.security.fat.common.SSO.clientTests/publish/servers/com.ibm.ws.security.SSO.op.wasReqUrl/bootstrap.properties:2]: Check copyrights on some of these. If the files originated in 2021, then that can stay. But if you're creating new files (even if they're based on other files), update the copyright to 2024.

**Review by ayoho — APPROVED** [review]

**Comment by ayoho** [comment]
#libby
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oidc.client_fat.3,com.ibm.ws.security.social_fat.LibertyOP.3

**Pull request merged**